dagobah/core/job.py

```diff
@@ -474,3 +477,55 @@ def destroy_snapshot(self):
         """ Destroy active copy of the snapshot """
         logger.debug('Destroying DAG snapshot for job {0}'.format(self.name))
         self.snapshot = None
+
+    def verify(self, context=None):
+        """
+        Verify that the job has no cycles where a JobTask circularly
+        references another JobTask so that we know we can safely snapshot
+        the DAG.
+        Explanation:
+            1. If the current job is in the context, the job is not valid
+            2. Perform a topological sort without expanding tasks (current job
+               is acyclic)
+            3. Traverse nodes in topological order. For each node that is a
+               Job, run verify on that node, passing in the current context.
+        """
+        # Check if job is not in current context, then add it
+        logger.info("Verifying DAG for {0}".format(self.name))
+        if context is None:
+            logger.debug("No context set, using empty set")
+            context = set()
+        if self.name in context:
+            logger.warn("Cycle detected: Job {0} already in context: {1}"
+                        .format(self.name, str(context)))
+            return False
+        context.add(self.name)
+        logger.debug("Context is now {0}".format(context))
+
+        # Topological sort verifies DAG is acyclic before digging deeper
+        topo_sorted = [self.tasks[t] for t in self.topological_sort()]
+
+        # Traverse nodes and verify any JobTasks
+        logger.debug("Traversing topologically sorted tasks.")
+        for task in topo_sorted:
+            if self.implements_expandable(task):
```

Comment:
I want to change this to the negative check, and do a continue in that case, which will eliminate 1 level of indentation from the rest of the following code

ex:

```python
if not self.implements_expandable(task):
    continue

logger.debug("Found expandable task: {0}".format(task.name))
cur_job = self.parent._resolve_job(task.target_job_name)
... 
```